recipe/build.sh

```diff
@@ -1,11 +1,84 @@
-export CARGO_FEATURE_EXTERNAL_CPP=1
-export PSP_ROOT_DIR=$SRC_DIR/perspective-cpp
-cd perspective_python-${PKG_VERSION}
-cp -r ../perspective_python-${PKG_VERSION}.data .
-cd rust/perspective-client
-export CARGO_FEATURE_EXTERNAL_PROTO=1
-cargo build
-cd ../../
-unset CARGO_FEATURE_EXTERNAL_PROTO
-${PYTHON} -m pip install . -vv
-${PYTHON} ${RECIPE_DIR}/copy.py
+#!/bin/bash
+set -e
+
+# XXX(tom): don't understand why it's needed to set PKG_CONFIG_PATH.
+# this directory is already present in pkg-config's config path as reported by
+# `pkg-config --variable pc_path pkg-config` in a --debug build environment.
+# and yet `pkg-config <some-dependency>` fails
+export PKG_CONFIG_PATH=$BUILD_PREFIX/lib/pkgconfig/:$PKG_CONFIG_PATH
+
+case "${target_platform}" in
+  linux-aarch64|osx-arm64)
+    arch="aarch64"
+    ;;
+  *)
+    arch="x86_64"
+    ;;
+esac
+export PSP_ARCH=$arch
+
+pnpm install --filter '@finos/perspective-python'
+
+# Boost!
+# Work around boost's inability to detect cxx compiler from $CXX.
+# In short: conda provides a compiler binary (compiler('cxx')) on the PATH and
+# sets CXX in the environment to point to it. On Linux, the compiler (basename
+# $CXX) is called something like x86_64-conda-linux-gnu-g++, and not g++.  This
+# conflicts with Boost's bootstrap script, which looks up g++ on the PATH when
+# toolchain=gcc is set.
+# So on Linux, we have to create a g++ binary on the PATH which point to $CXX.
+# NOTE: boost-feedstock does this more rigourously, writing out a
+# site-config.jam file and also patching the boost source to give more control
+# over the toolchain.  See build.sh and patches/ in
+# https://github.com/conda-forge/boost-feedstock/tree/91de9469b5564fd816e4298d217990481c280393/recipe
+#
+# Would be great if we could just use libboost-headers as a host dependency,
+# and not need to build Boost from source and statically link to it.  Thought we
+# needed to link Boost for arrow reasons, but arrow-cpp recipe only depends on libboost-headers
+# so maybe we can too somehow.
+# Note: assumes GNU mktemp, available from coreutils package
+psp_bin=$(mktemp -d -t 'psp-bin-XXXXXX')
+case "${cxx_compiler}" in
+  gxx)
+    cxx_compiler_plus="g++"
+    ;;
+  *)
+    cxx_compiler_plus="${cxx_compiler}"
+  ;;
+esac
+
+ORIG_PATH=$PATH
+export PATH=$psp_bin:$PATH
+ln -s $CXX $psp_bin/cxx
+ln -s $CXX $psp_bin/$cxx_compiler_plus
+
+# Install boost:
+# This configures both the install directory for install_tools.mjs,
+# and also the place where FindBoost looks for Boost.
+# Note: assumes GNU mktemp
+export Boost_ROOT=$(mktemp -d -t 'psp-boost-root-XXXXXX')
+node tools/perspective-scripts/install_tools.mjs
```

Comment:
Should never be running the install tools script in conda land, get from conda.

Reply:
Believe me, with all the rigamarole it took to make the b2 bootstrap script work, I would love to just depend on `libboost-headers`.

The word on the street was that perspective only needs to link Boost for something in the vendored arrow build. But I noticed that the arrow-cpp feedstock recipe only depends on `libboost-headers`; it doesn't link boost, not even dynamically. So maybe perspective can get away with just boost headers too. That would be great.

Reply:
Weird. Even though the arrow-cpp recipe only declares a dependency on `libboost-headers`, it does ends up bringing in `libboost` through its dependency on libthrift. Curiously though, neither libthrift.so nor libarrow*.so in this linux-64 environment  actually link any of the various libboost*.so.

```
$ conda create -n arrow-test arrow-cpp
$ conda activate arrow-test
$ ldd $CONDA_PREFIX/lib/libarrow*.so | grep libboost
# (nada)
$ ldd $CONDA_PREFIX/lib/libthrift*.so | grep libboost
# (nada)
```

Arrow's usage of boost looks smaller than what I remember. Every boost `#include` here, excluding tests, examples, and test support code, is header-only. Like, it's just boost/crc.hpp, right? https://github.com/search?q=repo%3Aapache%2Farrow+boost%2F+language%3AC%2B%2B&type=code&l=C%2B%2B

Reply:
WIP attempt at switching to libboost-headers: https://github.com/conda-forge/perspective-feedstock/commit/cad360878a28604909290254e78294c2558aaf9b

encountered some tomfoolery with Boost_INCLUDE_DIR disappearing from include_directories(), don't have time to look at it today. Will address when I'm back Monday

Reply:
> encountered some tomfoolery with Boost_INCLUDE_DIR disappearing from include_directories(), don't have time to look at it today

This went away as soon as I re-applied the `fix-flags.diff` patch, which (among other things) removes the resetting of CMAKE_CXX_FLAGS that upstream's build does (something which I don't understand the reasoning for well, myself). Something smells funny here - maybe something changed in more recent CMake versions?

In any case, I'd really like to contribute something upstream which switches as much as possible from CMAKE_CXX_FLAGS to target_include_directories, target_compiler_options, etc. This quest may be a quixotic one though

Reply:
There was a CMake cleanup PR, alas it was largely ignored but it fixed a lot of flag clobbering